packages/ses/types.d.ts

```diff
@@ -113,17 +113,51 @@ export interface EvaluateOptions {
   __rejectSomeDirectEvalExpressions__?: boolean;
 }
 
-// The DetailsToken is an empty object literal.
+/**
+ * A call to the `details` template literal makes and returns a fresh details
+ * token, which is a frozen empty object associated with the arguments of that
+ * `details` template literal expression.
+ */
 export type DetailsToken = Record<any, never>;
+/** Either a plain string, or made by the `details` template literal tag. */
 export type Details = string | DetailsToken;
 
 export interface AssertMakeErrorOptions {
+  /**
+   * Does not affect the error.name property. That remains determined by
+   * the constructor. Rather, the `errorName` determines how this error is
+   * identified in the causal console log's output.
+   */
   errorName?: string;
+
+  /**
+   * Discloses the error that caused this one, typically from a lower
+   * layer of abstraction. This is represented by a public `cause` data property
+   * on the error, not a hidden annotation.
+   */
   cause?: Error;
+
+  /**
+   * Normally only used when the ErrorConstuctor is `AggregateError`, to
+   * represent the set of prior errors aggregated together in this error,
+   * typically by `Promise.any`. But `makeError` allows it on any error.
+   * This is represented by a public `errors` data property on the error,
+   * not a hidden annotation.
+   */
   errors?: Error[];
+
+  /**
+   * Defaults to true. If true, `makeError` will apply `sanitizeError`
+   * to the error before returning it. See the comments on
+   * {@link sanitizeError}.
+   */
   sanitize?: boolean;
 }
 
+// Type all the overloads of the assertTypeof function.
+// There may eventually be a better way to do this, but
+// thems the breaks with Typescript 4.0.
```

Comment:
I think we can do this now
```suggestion
// TODO inline overloading
```

Reply:
Done